### DISCUSSION TITLE
Issue with configuration files Virtual Mixer

### DISCUSSION THREAD

**Discussion by LeidyMoraV** [PlusToolkit/PlusLib#1168] [Q&A] [answered]
Hi, I currently have the Ascension Trackstar, Siemens P300 ultrasound, and the Epiphan frame grabber VGA2USB, and I am trying to link all three hardware through PLUS to Slicer. Individually, through the 'Epiphan video capture device' and 'Ascension 3DG tracker' configurations, I am able to make the sensor probes and captures ultrasound screen appear through OpenIGT on Slicer. But, I cannot make them appear together in Slicer since I can only connect to 1 server/configuration at a time.

I've tried creating a new config file and loading it to launch server but it fails

`<PlusConfiguration version="2.0">

  <DataCollection StartupDelaySec="1.0">
    <DeviceSet 
      Name="PlusServer: Ephiphan and Ascension3D (Reference, Probe, Stylus)"
      Description="Prueba Ephiphan and Ascension 3D" />

    <Device
      Id="TrackerDevice"
      Type="Ascension3DG"
      FilterAcWideNotch="1"
      ToolReferenceFrame="Tracker"
      BaudRate="115200">
      <DataSources>
        <DataSource Type="Tool" Id="Reference" BufferSize="500" PortName="0" AveragedItemsForFiltering="20" />
        <DataSource Type="Tool" Id="Probe" BufferSize="500" PortName="1" AveragedItemsForFiltering="20" />
        <DataSource Type="Tool" Id="Stylus" BufferSize="500" PortName="2" AveragedItemsForFiltering="20" />
      </DataSources>
      <OutputChannels>
        <OutputChannel Id="TrackerStream">
          <DataSource Id="Reference" />
          <DataSource Id="Probe" />
          <DataSource Id="Stylus" />
        </OutputChannel>
      </OutputChannels>
    </Device>
    <Device
      Id="VideoDevice" 
      Type="Epiphan" >
      <DataSources>
        <DataSource Type="Video" Id="Video" PortUsImageOrientation="MF"  ImageType="RGB_COLOR"  />
      </DataSources>      
      <OutputChannels>
        <OutputChannel Id="VideoStream" VideoDataSourceId="Video" />
      </OutputChannels>
    </Device>
    <Device 
      Id="TrackedVideoDevice" 
      Type="VirtualMixer" >
      <InputChannels>
        <InputChannel Id="TrackerStream" />
        <InputChannel Id="VideoStream" />
      </InputChannels>
      <OutputChannels>
        <OutputChannel Id="TrackedVideoStream"/>
      </OutputChannels>
    </Device>   
    <Device
      Id="CaptureDevice"
      Type="VirtualCapture"
      BaseFilename="RecordingTest.igs.mha"
      EnableCapturingOnStart="FALSE" >
      <InputChannels>
        <InputChannel Id="TrackedVideoStream" />
      </InputChannels>
    </Device>
  </DataCollection>

  <CoordinateDefinitions>
    <Transform From="Image" To="Reference"
      Matrix="
        0.2 0.0 0.0 0.0
        0.0 0.2 0.0 0.0
        0.0 0.0 0.2 0.0        
        0 0 0 1" />
  </CoordinateDefinitions>
  
  <PlusOpenIGTLinkServer 
    MaxNumberOfIgtlMessagesToSend="1" 
    MaxTimeSpentWithProcessingMs="50" 
    ListeningPort="18944" 
    SendValidTransformsOnly="true" 
    OutputChannelId="TrackedVideoStream" > 
    <DefaultClientInfo> 
      <MessageTypes> 
        <Message Type="IMAGE" />
        <Message Type="TRANSFORM" />
      </MessageTypes>
      <ImageNames>
        <Image Name="Image" EmbeddedTransformToFrame="Reference" />
      </ImageNames>
      <TransformNames>
        <Transform Name="ProbeToTracker" />
        <Transform Name="ReferenceToTracker" />
        <Transform Name="StylusToTracker" />
      </TransformNames>
    </DefaultClientInfo>
  </PlusOpenIGTLinkServer>

</PlusConfiguration>
`

**Comment by adamrankin** [comment]
What does the log say when you try to connect?

**Reply by LeidyMoraV** [reply]
Connection failed!

Please select another device set and try again!

**Reply by adamrankin** [reply]
That is the status message in the main box.

Could you post the log contents by clicking the red/orange/green circle in the bottom right?

**Reply by LeidyMoraV** [reply]
```
|INFO|000.036000| Software version: Plus-2.8.0.62873a16 - Win32| in E:\D\PSNPb\PlusApp\PlusServerLauncher\PlusServerLauncherMainWindow.cxx(122)
|INFO|000.036000| Logging at level 3 to file: C:/Program Files (x86)/PlusApp-2.8.0.20191105-Win32/data/030824_153049_PlusLog.txt| in E:\D\PSNPb\PlusApp\PlusServerLauncher\PlusServerLauncherMainWindow.cxx(123)
|INFO|000.394000| Supported devices: - 3dConnexion (ver: Plus-2.8.0) - Ascension3DG (ver: Plus-2.8.0) - AuroraTracker (ver: NDICAPI-1.7) - BkProFocusOem (ver: Plus-2.8.0) - BrachyTracker (ver: Plus-2.8.0) - CertusTracker (ver: Optotrak API (Win32 DLL Version) Version 3.01.03) - ChRobotics (ver: Plus-2.8.0) - Epiphan (ver: Plus-2.8.0) - FakeTracker (ver: Plus-2.8.0) - GenericSerialDevice (ver: Plus-2.8.0) - ICCapturing (ver: The Imaging Source UDSHL-3.4) - ImageProcessor (ver: Plus-2.8.0) - IntelRealSense (ver: Plus-2.8.0) - Microchip (ver: Plus-2.8.0) - MicronTracker (ver: MTC-3.7.6.8) - MmfVideo (ver: Plus-2.8.0) - NDITracker (ver: NDICAPI-1.7) - NoiseVideo (ver: Plus-2.8.0) - OpenCVVideo (ver: Plus-2.8.0) - OpenIGTLinkTracker (ver: OpenIGTLink v3.1.0) - OpenIGTLinkVideo (ver: OpenIGTLink v3.1.0) - OptiTrack (ver: Plus-2.8.0) - OpticalMarkerTracker (ver: Plus-2.8.0) - PhidgetSpatial (ver: Plus-2.8.0) - PolarisTracker (ver: NDICAPI-1.7) - SavedDataSource (ver: Plus-2.8.0) - SonixPortaVideo (ver: UltrasonixSDK-6.1.1) - SonixVideo (ver: UltrasonixSDK-6.1.1) - USDigitalEncodersTracker (ver: Plus-2.8.0) - UsSimulator (ver: Plus-2.8.0) - VFWVideo (ver: Plus-2.8.0) - VirtualBufferedCapture (ver: Plus-2.8.0) - VirtualCapture (ver: Plus-2.8.0) - VirtualDiscCapture (ver: Plus-2.8.0) - VirtualMixer (ver: Plus-2.8.0) - VirtualSwitcher (ver: Plus-2.8.0) - VirtualTextRecognizer (ver: Plus-2.8.0) - VirtualVolumeReconstructor (ver: Plus-2.8.0) - WitMotionTracker (ver: Plus-2.8.0) | in E:\D\PSNPb\PlusApp\PlusServerLauncher\PlusServerLauncherMainWindow.cxx(127)
|INFO|000.399000| Server host name: E5-ME-DACO12| in E:\D\PSNPb\PlusApp\PlusServerLauncher\PlusServerLauncherMainWindow.cxx(158)
|INFO|000.412000| Server host domain: ad.psu.edu| in E:\D\PSNPb\PlusApp\PlusServerLauncher\PlusServerLauncherMainWindow.cxx(161)
|INFO|000.414000| Server IP addresses: 130.203.244.194, 127.0.0.1| in E:\D\PSNPb\PlusApp\PlusServerLauncher\PlusServerLauncherMainWindow.cxx(178)
|INFO|000.414000| Start remote control server at port: 18904| in E:\D\PSNPb\PlusApp\PlusServerLauncher\PlusServerLauncherMainWindow.cxx(187)
|ERROR|009.744000| Failed to open C:/Program Files (x86)/PlusApp-2.8.0.20191105-Win32/bin/PlusConfig.xml for writing| in E:\D\PSNPb\PlusLib\src\PlusCommon\PlusCommon.cxx(527)
|ERROR|032.765000| Failed to open C:/Program Files (x86)/PlusApp-2.8.0.20191105-Win32/bin/PlusConfig.xml for writing| in E:\D\PSNPb\PlusLib\src\PlusCommon\PlusCommon.cxx(527)
|INFO|032.766000| Connect using configuration file: C:\Users\lfm5705\Documents\PlusDeviceSet_Ascension3DG and Ephipan.xml| in E:\D\PSNPb\PlusApp\PlusServerLauncher\PlusServerLauncherMainWindow.cxx(639)
|INFO|032.766000| Server process command line: "C:/Program Files (x86)/PlusApp-2.8.0.20191105-Win32/bin/PlusServer.exe" --config-file="PlusDeviceSet_Ascension3DG and Ephipan.xml" --verbose=3| in E:\D\PSNPb\PlusApp\PlusServerLauncher\PlusServerLauncherMainWindow.cxx(320)
|INFO|033.413000| Server process started successfully| in E:\D\PSNPb\PlusApp\PlusServerLauncher\PlusServerLauncherMainWindow.cxx(329)
|INFO|044.168000|SERVER> System start timestamp: 2038040
|ERROR|044.171000|SERVER> Failed to open C:/Program Files (x86)/PlusApp-2.8.0.20191105-Win32/bin/PlusConfig.xml for writing| in :\D\PSNPb\PlusLib\src\PlusCommon\PlusCommon.cxx(527)
|INFO|044.171000|SERVER> Software version: Plus-2.8.0.62873a16 - Win32
|INFO|044.171000|SERVER> Logging at level 3 (INFO) to file: C:/Program Files (x86)/PlusApp-2.8.0.20191105-Win32/data/030824_153134_PlusLog.txt
|ERROR|044.172000|SERVER> Reading device set configuration file failed: PlusDeviceSet_Ascension3DG and Ephipan.xml does not exist in the current directory or in C:/Program Files (x86)/PlusApp-2.8.0.20191105-Win32/config| in :\D\PSNPb\PlusLib\src\PlusCommon\vtkPlusConfig.cxx(233)
|ERROR|044.172000|SERVER> Datacollector failed to read configuration| in :\D\PSNPb\PlusLib\src\PlusServer\Tools\PlusServer.cxx(87)
|ERROR|044.181000| Server stopped unexpectedly. Return code: 1| in E:\D\PSNPb\PlusApp\PlusServerLauncher\PlusServerLauncherMainWindow.cxx(864)
|INFO|044.181000| Disconnect request successful| in E:\D\PSNPb\PlusApp\PlusServerLauncher\PlusServerLauncherMainWindow.cxx(632)
```

**Reply by adamrankin** [reply, marked as answer]
It looks like you have some file permission issues. You may want to re-install Plus to your user folder

**Reply by LeidyMoraV** [reply]
Thanks! I tried with admin access and it worked
